pkg/dependencies/factory.go

```diff
@@ -250,6 +257,16 @@ func (f *Factory) WithExecutableBuilder() *Factory {
 		}
 
 		if f.executablesConfig.useDockerContainer {
+
+			if f.registryAuth {
+				username := os.Getenv("REGISTRY_USERNAME")
+				password := os.Getenv("REGISTRY_PASSWORD")
+				err := f.executablesConfig.dockerClient.Login(context.Background(), f.registryMirror, username, password)
+				if err != nil {
+					return err
+				}
+			}
```

Comment:
This doesn't seem like the right place to do a docker login. I wonder if we can do this earlier? Perhaps it even requires an explicit step high up in the call path?